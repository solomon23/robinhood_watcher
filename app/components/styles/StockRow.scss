.stockRow {
  text-align: left;
  padding: 8px 16px;
  color: #fff;
  font-size: 20px;

  & > div {
    display: inline-block;
  }

  &:hover {
    background-color: #212125;
    cursor: pointer;
  }
}

.count {
  font-size: 12px;
  margin-top: 2px;
}

.value {
  font-size: 22px;
  float: right;
  color: green;
  margin-top: 4px;
}

.loss {
  color: red;
}
