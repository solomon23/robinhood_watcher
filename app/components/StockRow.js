// @flow
import React, { Component } from 'react'
import pluralize from 'pluralize'
import classnames from 'classnames'
import { ipcRenderer } from 'electron'
import { USD } from '../services/utils'
import styles from './styles/StockRow.scss'

type Props = {
  stock: object
}

export default class StockList extends Component<Props> {
  props: Props

  static onClick(symbol, stock) {
    ipcRenderer.send('CHART', { symbol, stock })
  }

  render() {
    const { stock } = this.props

    const price = stock.quote.last_extended_hours_trade_price || stock.quote.last_trade_price
    const oldPrice = stock.quote.previous_close

    const dif = (price - oldPrice)
    const difference = `${dif > 0 ? '+' : ''}${USD(stock.quantity * dif)}`

    return (
      <div className={styles.stockRow} onClick={() => StockList.onClick(stock.symbol, stock)}>
        <div className={styles.sybmol}>
          {stock.symbol}
          <div className={styles.count}>{Number(stock.quantity)} {pluralize('Share', stock.quantity)}</div>
        </div>
        <div className={classnames(styles.value, { [styles.loss]: dif < 0 })}>
          {difference}
        </div>
      </div>
    )
  }
}
