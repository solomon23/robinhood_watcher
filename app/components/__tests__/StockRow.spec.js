import React from 'react'
import Enzyme, { shallow } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import { shallowToJson } from 'enzyme-to-json'
import StockRow from '../StockRow'

Enzyme.configure({ adapter: new Adapter() })

const actions = {}
const defaultState = {
  stock: {},
}

describe('StockRow component', () => {
  beforeEach(() => {
    defaultState.stock = { 'symbol': 'a', quantity: 1, quote: { last_trade_price: 10, previous_close: 5 } }
  })

  it('should display positive value', () => {
    const component = shallow(<StockRow {...defaultState} {...actions} />)
    expect(shallowToJson(component)).toMatchSnapshot()
  })

  it('should display negative value', () => {
    defaultState.stock.quote.previous_close = 20
    const component = shallow(<StockRow {...defaultState} {...actions} />)
    expect(shallowToJson(component)).toMatchSnapshot()
  })

  it('should display after hours value', () => {
    defaultState.stock.quote.last_extended_hours_trade_price = 20
    const component = shallow(<StockRow {...defaultState} {...actions} />)
    expect(shallowToJson(component)).toMatchSnapshot()
  })
})
