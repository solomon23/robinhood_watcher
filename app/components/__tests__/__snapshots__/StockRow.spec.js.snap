// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StockRow component should display after hours value 1`] = `
<div
  className="stockRow"
  onClick={[Function]}
>
  <div
    className="sybmol"
  >
    a
    <div
      className="count"
    >
      1
       
      Share
    </div>
  </div>
  <div
    className="value"
  >
    +$15.00
  </div>
</div>
`;

exports[`StockRow component should display negative value 1`] = `
<div
  className="stockRow"
  onClick={[Function]}
>
  <div
    className="sybmol"
  >
    a
    <div
      className="count"
    >
      1
       
      Share
    </div>
  </div>
  <div
    className="value loss"
  >
    -$10.00
  </div>
</div>
`;

exports[`StockRow component should display positive value 1`] = `
<div
  className="stockRow"
  onClick={[Function]}
>
  <div
    className="sybmol"
  >
    a
    <div
      className="count"
    >
      1
       
      Share
    </div>
  </div>
  <div
    className="value"
  >
    +$5.00
  </div>
</div>
`;
